packages/dashboard/src/app/views/myStories/filters/taxonomy/useTaxonomyFilters.js

```diff
@@ -0,0 +1,151 @@
+};
+
+export const getFilterSearchResultsLabel = (filter) => {
+  return (
+    filter.labels?.searchItems ||
+    (filter.labels?.name ? `Search ${filter.labels.name}` : 'Search Taxonomy')
+  );
+};
+
+/**
+ * Hook used for taxonomy filters logic
+ * Initilizes the taxonomy filters data
+ *
+ * @return {Array} [taxonomies, queryTaxonomyTerm] initial taxonomy filters data and a function to query the terms
+ */
+
+function useTaxonomyFilters() {
+  const [taxonomies, setTaxonomies] = useState([]);
+
+  const { getTaxonomies, getTaxonomyTerm } = useApi(
+    ({
+      actions: {
+        taxonomyApi: { getTaxonomies, getTaxonomyTerm },
+      },
+    }) => ({
+      getTaxonomies,
+      getTaxonomyTerm,
+    })
+  );
+
+  /**
+   * Query individual taxonomy data
+   * This is needed to initilize the primaryOptions and needed to search and set queriedOptions
+   *
+   * @param {Object} taxonomy object holding taxonomy data, including restBase and restPath used to fetch individual taxonomy terms
+   * @param {string|undefined} search string use to query taxonomy by name
+   * @return {Array} taxonomy terms
+   */
+  const queryTaxonomyTerm = useCallback(
+    async (taxonomy, search) => {
+      const { restBase, restPath } = taxonomy;
+      const terms = await getTaxonomyTerm(restPath, { search });
+      return terms.map((t) => ({
+        ...t,
+        restBase,
+        restPath,
+      }));
+    },
+    [getTaxonomyTerm]
+  );
+
+  /**
+   * Query all the taxonomies
+   * This should only be needed once get all the taxonomies and set individual term data
+   *
+   * @see queryTaxonomyTerm
+   */
+  const queryTaxonomies = useCallback(async () => {
+    const data = await getTaxonomies();
+    const promises = [];
+    const hierarchicalTaxonomies = data.filter(({ hierarchical }) =>
+      Boolean(hierarchical)
+    );
+
+    for (const taxonomy of hierarchicalTaxonomies) {
+      // initilize the data with an empty array
+      taxonomy.data = [];
+      promises.push(queryTaxonomyTerm(taxonomy));
+    }
+
+    const fetched = await Promise.all(promises);
+
+    for (const arr of fetched) {
+      // grab the first elements 'taxonomy' to map the array of terms back to the parent taxonomy
+      const key = arr.at(0)?.taxonomy;
+      if (key) {
+        const taxonomy = hierarchicalTaxonomies.find((h) => h.slug === key);
+        if (taxonomy) {
+          taxonomy.data = arr;
+        }
+      }
+    }
+
+    setTaxonomies(hierarchicalTaxonomies);
+  }, [setTaxonomies, getTaxonomies, queryTaxonomyTerm]);
+
+  const initializeTaxonomyFilters = useCallback(() => {
+    return taxonomies.map((taxonomy) => ({
+      key: taxonomy.restBase,
+      restPath: taxonomy.restPath,
+      labels: taxonomy.labels,
+      filterId: null,
+      primaryOptions: taxonomy.data,
+      queriedOptions: taxonomy.data,
+      query: queryTaxonomyTerm,
+      placeholder: getFilterPlaceholder(taxonomy),
+      ariaLabel: getFilterAriaLabel(taxonomy),
```

Comment:
```suggestion
      ariaLabel: taxonomy.labels.filterByItem,
```

Reply:
🤔 can we assume these 'labels' are going to be on every taxonomy?

Reply:
Yes 👍